design/20200331.acme-self-check-failure-policy.md

```diff
@@ -0,0 +1,89 @@
+---
+title: Failure policy for ACME self checks
+authors:
+  - "@anton-johansson"
+reviewers: []
+approvers: []
+creation-date: 2020-03-31
+last-updated: 2020-03-31
+status: ?
+---
+
+# Failure policy for ACME self checks
+
+<!-- toc -->
+- [Summary](#summary)
+- [Motivation](#motivation)
+  - [Goals](#goals)
+  - [Non-Goals](#non-goals)
+- [Proposal](#proposal)
+  - [API Changes](#api-changes)
+  - [Controller Behaviour](#controller-behaviour)
+  - [Internal API Resource Behaviour](#internal-api-resource-behaviour)
+  - [Test Plan](#test-plan)
+  - [Risks and Mitigations](#risks-and-mitigations)
+  - [Graduation Criteria](#graduation-criteria)
+      - [Alpha](#alpha)
+      - [Alpha -&gt; Beta Graduation](#alpha---beta-graduation)
+      - [Beta -&gt; GA Graduation](#beta---ga-graduation)
+      - [Removing a deprecated flag](#removing-a-deprecated-flag)
+  - [Version Skew Strategy](#version-skew-strategy)
+<!-- /toc -->
+
+## Summary
+
+When requesting certificates from Lets Encrypt, cert-manager performs a self-check to make sure that we properly own the domain. This is to avoid redundant calls to Lets Encrypt and avoid getting rate limited. This is very good safefty net. However, it would be a great addition to cert-manager if this check could be disabled or at least configured to allow failure.
```

Comment:
> to make sure that we properly own the domain

"to make sure that we have properly configured the ACME challenge response for the domain"